### ISSUE TITLE
Show `ci-link` for PR next to it's title/number

### ISSUE THREAD

**Issue by ofirnk** [sindresorhus/refined-github#4086]
Similarly to repo CI status:
https://github.com/sindresorhus/refined-github/blob/main/source/features/ci-link.tsx
![image](https://user-images.githubusercontent.com/8449713/110773001-ebd6de80-8264-11eb-8684-278cb6b118f0.png)

I propose adding a CI status next to PR name/title/number.
Reasons:
1. On long PRs - you need to scroll A LOT to see CI status ([for example](https://github.com/sindresorhus/refined-github/pull/4023) )
2. When you're in another tab - like changes - you need to switch tabs to know CI status. So you need to switch tab to conversations + scroll to bottom.
3. When you're already scrolled in changes view - you need to *lose context* to know CI status. Like half way thru review, or [direct link](https://github.com/sindresorhus/refined-github/pull/3810/files#diff-5a831ea67cf5cf8703b0de46901ab25bd191f56b320053be9332d9a3b0d01d15R224)

**Comment by ofirnk** [comment]
I suggest to attach it to PR title - like that (expanded)
![image](https://user-images.githubusercontent.com/8449713/110774247-41f85180-8266-11eb-89f5-9e4fc7f6917b.png)

**Comment by ofirnk** [comment]
And like that (scrolled)
![image](https://user-images.githubusercontent.com/8449713/110775009-1b86e600-8267-11eb-9e8e-2f937a618e5a.png)

**Comment by kidonng** [comment]
Yeah, I like how GitHub already does this on the mobile app:

![Screenshot_20210313202643.png](https://user-images.githubusercontent.com/44045911/111029980-c56a9c00-843a-11eb-83aa-1d0458ac8ac5.png)

**Comment by yakov116** [comment]
For anyone that wants to do this see the feedback on https://github.com/sindresorhus/refined-github/pull/4100

**Comment by yakov116** [comment]
I do not think this is possible anymore as GitHub stopped showing CI links on fork commit list see https://github.com/sindresorhus/refined-github/pull/4415/commits

**Comment by fregante** [comment]
Fantastic… 

but maybe we can find their URL anyway if it still works

**Comment by kidonng** [comment]
Maybe just a extra request:

1. Get the latest commit if it's not available on the page (for example on Files tab)
2. Fetch the commit page and extract CI icon from it

**Comment by yakov116** [comment]
> 2\. Fetch the commit page and extract CI icon from it

What commit page are you referring to? Both in the PR and it the repo itself the ci-lint is not there.

**Comment by kidonng** [comment]
Something like https://github.com/sindresorhus/refined-github/pull/4415/commits/4a46492f66357d47197d3632b9b7a53690012eaf

**Comment by yakov116** [comment]
Ohh I see next to the title of the commit.

**Comment by fregante** [comment]
> Maybe just an extra request:

Not ideal, especially since we already can't cache it

**Comment by kidonng** [comment]
> I'd rather just have it on the Conversation tab for the time being, and then maybe add a `fetch` if we have the commit information on the page or if have the direct URL to the icon.

From what I can see, there are three situations (please correct if there's something wrong):

- On Conversations tab, the latest commit along with the icon is already presented (not always though? Not sure).   We don't need to make any request.
- On Commits tab, we have the latest commit but not the icon, so we need one request to get the icon.
- On Changes tab, we don't (?) have the latest commit info, so we need two requests, one for the commit list and one for the icon.

**Comment by ofirnk** [comment]
On changes we have the "view file" link which includes the latest commit hash.
On multiple commits I think there's "view changes at/between commits"

**Comment by ofirnk** [comment]
> Something like [this](https://github.com/sindresorhus/refined-github/pull/4415/commits/4a46492f66357d47197d3632b9b7a53690012eaf) or [this](https://github.com/sindresorhus/refined-github/commit/4a46492f66357d47197d3632b9b7a53690012eaf).

There seems to be better way thru this URL:

https://github.com/sindresorhus/refined-github/pull/4023/commits/4a46492f66357d47197d3632b9b7a53690012eaf/_render_node/commit/pull_condensed
which returns
![image](https://user-images.githubusercontent.com/8449713/120324414-4412ff00-c2ef-11eb-9d56-4effb6269d39.png)

**Comment by fregante** [comment]
This could be more easily implemented using:

- `getCiDetails` added in https://github.com/refined-github/refined-github/pull/5836
- `getPrInfo` added in https://github.com/refined-github/refined-github/pull/5768

Issues:

- `getPrInfo` actually needs to also get _the latest commit_ of the PR, not just the base commit
- the latest commit can update multiple times while the tab is open, so we also need to replace the icon

But actually I suggest:

- showing `pr-ci-link` **only on the Conversation tab** so we have access to the latest commit by looking on the page
- listening to `onNewComments` to check whether we need to replace the icon if new commits appear

This would make it relatively easy(er) to implement.

**Comment by fregante** [comment]
But actually the solution in https://github.com/refined-github/refined-github/pull/4100 was pretty straightforward. Maybe it can be renewed using the latest adjustments to `ci-link` (the feature was rewritten in #5836)

**Comment by SunsetTechuila** [comment]
https://github.blog/changelog/2026-03-05-quick-access-to-merge-status-in-pull-requests-in-public-preview/

**Comment by fregante** [comment]
Indeed! It's a good addition

**Comment by SunsetTechuila** [comment]
Sometimes problems do resolve themselves if you wait long enough

**Comment by fregante** [comment]
If we wait long enough, Microsoft will shut down and Refined GitHub will follow 😁
